### PR TITLE
API keys: Add deprecation to api keys

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What is this feature?**
Adds deprecation notice for the create path of API keys w. a link to the new documentation outlining how to migrate your API calls to service accounts.

linking to page -https://grafana.com/docs/grafana/next/administration/api-keys/#migrate-api-keys-to-grafana-service-accounts-using-the-api

**notes**
Part of a bigger sunsetting of API keys - https://github.com/grafana/grafana/issues/53567

### CODE REVIEW

File: docs/sources/administration/api-keys/index.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ An API key is a randomly generated string that external systems use to interact
 
 When you create an API key, you specify a **Role** that determines the permissions associated with the API key. Role permissions control that actions the API key can perform on Grafana resources.
 
-> **Note:** If you use Grafana v8.5 or newer, use service accounts instead of API keys. For more information, refer to [Grafana service accounts]({{< relref "../service-accounts/" >}}).
+> **Note:** If you use Grafana v9.1 or newer, use service accounts instead of API keys. For more information, refer to [Grafana service accounts]({{< relref "../service-accounts/" >}}).
```
Comment (eleijonmarck) [inline docs/sources/administration/api-keys/index.md:20]: this is a fix for documenting the wrong version

File: pkg/api/apikey.go [hunk 2]
```diff
@@ -20,6 +20,11 @@ import (
 //
 // Will return auth keys.
 //
+// Deprecated: true.
+//
+// Deprecated please use GET /api/serviceaccounts and GET /api/serviceaccounts/{id}/tokens instead
```
Comment (IevaVasiljeva) [inline pkg/api/apikey.go:25]: super nit: I'd add a punctuation mark after `deprecated`. Something like `Deprecated: please use...` or `Deprecated. Please use...`.

**Review by IevaVasiljeva — APPROVED** [review]
Looks good 👍

File: pkg/api/apikey.go (hunk 2 above)
Comment (eleijonmarck) [reply pkg/api/apikey.go:25]: i do not think it's nit. this is about clear communication on this. and should be reviewed as such

**Review by mildwonkey — APPROVED** [review]
lgtm!

**Pull request merged**